launch_ros/launch_ros/actions/lifecycle_node.py

```diff
@@ -145,21 +92,30 @@ def execute(self, context: launch.LaunchContext) -> Optional[List[Action]]:
         self._perform_substitutions(context)  # ensure self.node_name is expanded
         if '<node_name_unspecified>' in self.node_name:
             raise RuntimeError('node_name unexpectedly incomplete for lifecycle node')
-        node = get_ros_node(context)
-        # Create a subscription to monitor the state changes of the subprocess.
-        self.__rclpy_subscription = node.create_subscription(
-            lifecycle_msgs.msg.TransitionEvent,
-            '{}/transition_event'.format(self.node_name),
-            functools.partial(self._on_transition_event, context),
-            10)
-        # Create a service client to change state on demand.
-        self.__rclpy_change_state_client = node.create_client(
-            lifecycle_msgs.srv.ChangeState,
-            '{}/change_state'.format(self.node_name))
-        # Register an event handler to change states on a ChangeState lifecycle event.
-        context.register_event_handler(launch.EventHandler(
-            matcher=lambda event: isinstance(event, ChangeState),
-            entities=[launch.actions.OpaqueFunction(function=self._on_change_state_event)],
-        ))
+
+        self.__lifecycle_event_manager = LifecycleEventManager(self)
+        self.__lifecycle_event_manager.setup_lifecycle_manager(context)
+
+        # If autostart is enabled, transition to the 'active' state.
+        autostart_actions = None
+        if self.node_autostart:
```

Comment:
Do you need to have a condition here for when autostart is `None`? Currently this condition is implicitly like `if self.node_autostart is not None and self.node_autostart is True`, which is maybe what you want, but also then why can it ever be `None`?

Reply:
I remove None optionality